subsys/lorawan/lorawan.c

```diff
@@ -495,6 +495,54 @@ int lorawan_set_datarate(enum lorawan_datarate dr)
 	return 0;
 }
 
+int lorawan_set_channel_mask(enum lorawan_channel_mask mask, bool set_default)
+{
+	MibRequestConfirm_t mib_req;
+
+	uint64_t pri_mask = 0;
+	uint32_t sec_mask = 0;
+	uint16_t channel_mask[6] = { 0 };
+
+	if (mask > LORAWAN_CHAN_32_63) {
+		/* 64 channels */
+		pri_mask = 0xFFFFFFFFFFFFFFFF;
+	} else if (mask > LORAWAN_CHAN_48_63) {
+		/* 32 channels */
+		pri_mask = 0xFFFFFFFF << ((mask - LORAWAN_CHAN_0_31) * 32);
+	} else if (mask > LORAWAN_CHAN_56_63) {
+		/* 16 channels */
+		pri_mask = 0xFFFF << ((mask - LORAWAN_CHAN_0_15) * 16);
+	} else {
+		/* 8 channels */
+		pri_mask = 0xFF << ((mask - LORAWAN_CHAN_0_7) * 8);
+		sec_mask = 1 << (mask - LORAWAN_CHAN_0_7);
+	}
+
+	channel_mask[0] = pri_mask & 0xFFFF;
+	channel_mask[1] = (pri_mask >> 16) & 0xFFFF;
+	channel_mask[2] = (pri_mask >> 32) & 0xFFFF;
+	channel_mask[3] = (pri_mask >> 48) & 0xFFFF;
+
+	channel_mask[4] = sec_mask & 0xFFFF;
+	channel_mask[5] = (sec_mask >> 16) & 0xFFFF;
```

Comment:
Just trying to understand what's going on here, does this kind of correspond to Section 2.2.5 of https://resources.lora-alliance.org/technical-specifications/lorawan-regional-parameters-v1-0-2rb with some custom masking for our own implementation?

Reply:
It's not custom. It's the channel masking that is required for the loramac library. Each bit corresponds directly to the lorawan channel number.